### PR TITLE
Feature/us004

### PR DESCRIPTION
**Pull request by kamil3197**

### CODE REVIEW

File: product-backend/scripts/create_tables.sql
Comment (leszekgu) [inline product-backend/scripts/create_tables.sql:1]: please change the sql to do alter table. Treat JobRoles as already existing table in the database that can't be removed only altered.
So the order of things in this script would be :
existing tables
Band tables
alteration of JobRoles table

File: product-backend/src/main/java/org/kainos/ea/db/JobRoleDao.java [hunk 1]
```diff
@@ -14,13 +14,15 @@ public class JobRoleDao {
...
         Statement st = c.createStatement();
-        ResultSet rs = st.executeQuery("SELECT job_role_id, job_role_title FROM JobRoles;");
+        ResultSet rs = st.executeQuery("SELECT JR.job_role_id,JR.job_role_title, B.level, B.name  FROM JobRoles JR JOIN Band B ON JR.band_id = B.id;");
         List<JobRole> jobRoleList = new ArrayList<>();
 
         while (rs.next()) {
             JobRole jobRole = new JobRole(
                     rs.getInt("job_role_id"),
-                    rs.getString("job_role_title")
+                    rs.getString("job_role_title"),
```
Comment (leszekgu) [inline product-backend/src/main/java/org/kainos/ea/db/JobRoleDao.java:23]: missing `JR.`

File: product-backend/src/main/java/org/kainos/ea/db/JobSpecificationDao.java [hunk 2]
```diff
@@ -15,15 +14,21 @@ public class JobSpecificationDao {
...
+        ResultSet rs = st.executeQuery("SELECT S.role_id, S.summary, S.sharepoint_link, JR.job_role_title, B.level, B.name\n" +
+                "        FROM Specifications S\n" +
+                "        INNER JOIN JobRoles JR ON S.role_id = JR.job_role_id\n" +
+                "        INNER JOIN Band B ON JR.band_id = B.id\n" +
+                "        WHERE S.id = " + role_id);
 
 
         if (rs.next()) {
             return Optional.of(new JobSpecification(
                     rs.getInt("role_id"),
```
Comment (leszekgu) [inline product-backend/src/main/java/org/kainos/ea/db/JobSpecificationDao.java:27]: missing `S.` ?

File: product-web-app/service/bandService.ts
Comment (leszekgu) [inline product-web-app/service/bandService.ts:1]: missing some simple unit tests: happy path, empty table, exception(s) from backend

File: product-backend/src/main/java/org/kainos/ea/api/BandService.java
Comment (leszekgu) [inline product-backend/src/main/java/org/kainos/ea/api/BandService.java:1]: missing unit tests for this class

**Review by leszekgu — APPROVED** [review]
